Improve copy to clipboard code

kamioon opened the pull request:
### Description

This is a PR for adding a **copy** to the clipboard on the export page that was requested (#18529). 

<img width="454" alt="phpmyadmin_copy_export" src="https://github.com/phpmyadmin/phpmyadmin/assets/3734100/6c0d0d24-56cd-4a87-9c3d-1e8648042d5e">

To keep the text format copied to the clipboard temporary `input` has changed to the `textarea`.

kamioon committed:
a copy to the clipboard button for text output on the export page was added

Signed-off-by: kamioon <kamran.azari@gmail.com>

williamdes reviewed (approved):
Very nice

kamioon committed:
remove unnecessarily comment

Signed-off-by: kamioon <kamran.azari@gmail.com>

williamdes commented on js/src/modules/functions.ts:
```diff
@@ -1516,8 +1516,13 @@ function dismissNotifications () {
+            var copyText = $(this).attr('data-text');
+            var copyTarget = $(this).attr('data-target');
+            if (copyTarget) {
+                copyText = String($(copyTarget).val());
```
```suggestion
                copyText = $(copyTarget).val();
```

kamioon committed:
the String object removed from the result of copy selector

Signed-off-by: kamioon <kamran.azari@gmail.com>

williamdes reviewed (approved):
💯

williamdes commented:
Can you check why tests are not passing?

kamioon committed:
a separate function for the copyExportBtn has been created and copy message moved to a new function.

Signed-off-by: kamioon <kamran.azari@gmail.com>

williamdes reviewed (changes requested):
what was wrong with the previous version?

kamioon commented:
`[tsl] ERROR in /home/runner/work/phpmyadmin/phpmyadmin/js/src/modules/functions.ts(1522,17)
      TS2322: Type 'string | number | string[]' is not assignable to type 'string'.`
@williamdes the `.val()` returns a mixed type and that was the reason I had added the `String()`. I've changed a little bit more. separated them into two different functions and moved the copy message to a new function. I think it's better now. otherwise, reverting the two latest commits could work fine but your commit suggestion wouldn't be applied there.

kamioon commented:
> what was wrong with the previous version?

Sorry, I just saw that you had approved the previous one. I explained here the reason for failing the test and I think the new changes are cleaner than using the `copyQueryBtn` for two purposes. (it should handle two different text sources.)

> `[tsl] ERROR in /home/runner/work/phpmyadmin/phpmyadmin/js/src/modules/functions.ts(1522,17) TS2322: Type 'string | number | string[]' is not assignable to type 'string'.` @williamdes the `.val()` returns a mixed type and that was the reason I had added the `String()`. I've changed a little bit more. separated them into two different functions and moved the copy message to a new function. I think it's better now. otherwise, reverting the two latest commits could work fine but your commit suggestion wouldn't be applied there.

williamdes commented on js/src/modules/functions.ts:
```diff
@@ -1516,17 +1516,28 @@ function dismissNotifications () {
+            var copyStatus = Functions.copyToClipboard($(copyTarget).val());
+            displayCopyStatus(this, copyStatus);
+        });
+
```
could you add a block of documentation like other functions to document the input and output
or much better: add typescript types

williamdes reviewed (approved):
Thank for explaining, looks okay
I left one minor comment about typing

kamioon committed:
the typescript types and documentation added for the displayCopyStatus

Signed-off-by: kamioon <kamran.azari@gmail.com>

williamdes reviewed (approved):
This really looks good
